### PR TITLE
[JENKINS-73339] Require Java 17 and Jetty 12 (EE 9) for plugin development

### LINKED ISSUE #918: Remove ASM workaround
When ASM is removed from Jenkins core and this change has propagated through a sufficient number of LTS releases, the workaround added in #916 can be removed.

### LINKED ISSUE #936: Plugin POM hard-codes EE 8
https://github.com/jenkinsci/plugin-pom/blob/8561e2e8c77c344d0465ec6c77e2e3eedc927c00/pom.xml#L182-L187 hard-codes the Servlet API to EE 8 (the last version with `javax` imports), which means that when core is upgraded to EE 9 and plugins upgrade their core baseline to that version, their tests will start failing, even if they have been otherwise prepared with a test harness that supports both EE 8 and EE 9 (see https://github.com/jenkinsci/jenkins-test-harness/pull/770). This manifests in strange and hard-to-debug errors, like not being able to import `jakarta` servlet types, or the compatibility methods introduced in https://github.com/jenkinsci/javax-servlet-api/pull/5 not existing (because the wrong `javax` version of the `jakarta.servlet-api` takes precedence over the package delivering the compatibility methods).

The ideal solution is to simply delete these lines and let the version be managed by the core (not plugin) BOM, which is possible as of https://github.com/jenkinsci/jenkins/pull/9288; however, it would imply raising the minimum core version supported by this repository to 2.459 or higher, something that is likely too aggressive at this point in time.

### PR DESCRIPTION
**Pull request by basil**
In draft until EE 9 is delivered in an LTS release.

### Testing done

`mvn clean verify`

Fixes #936
Fixes #918

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -74,13 +74,13 @@
     <frontend-version>1.15.0</frontend-version>
     <gmavenplus-plugin.version>3.0.2</gmavenplus-plugin.version>
     <hamcrest.version>3.0</hamcrest.version>
-    <hpi-plugin.version>3.57</hpi-plugin.version>
+    <hpi-plugin.version>3.58</hpi-plugin.version>
     <incrementals-enforce-minimum.version>1.0-beta-4</incrementals-enforce-minimum.version>
     <incrementals-plugin.version>1.8</incrementals-plugin.version>
     <jacoco-maven-plugin.version>0.8.12</jacoco-maven-plugin.version>
-    <jenkins.version>2.361</jenkins.version>
+    <jenkins.version>2.476</jenkins.version>
```
Comment (basil) [inline pom.xml:81]: Currently using 2.476 in this PR as a placeholder, but planning to update this to whatever is selected as the next LTS line.

File: pom.xml [hunk 2]
```diff
@@ -74,13 +74,13 @@
...
     <jenkins-bom.version>${jenkins.version}</jenkins-bom.version>
-    <jenkins-test-harness.version>2225.2230.v6210cb_b_827f9</jenkins-test-harness.version>
+    <jenkins-test-harness.version>2272.v5db_817b_a_6f0d</jenkins-test-harness.version>
```
Comment (basil) [inline pom.xml:83]: Requires Java 17 or newer and Jetty 12 (EE 9) or newer.

File: pom.xml [hunk 3]
```diff
@@ -74,13 +74,13 @@
     <frontend-version>1.15.0</frontend-version>
     <gmavenplus-plugin.version>3.0.2</gmavenplus-plugin.version>
     <hamcrest.version>3.0</hamcrest.version>
-    <hpi-plugin.version>3.57</hpi-plugin.version>
+    <hpi-plugin.version>3.58</hpi-plugin.version>
```
Comment (basil) [inline pom.xml:77]: Requires Java 17 or newer and Jetty 12 (EE 9) or newer.

**Review by jglick — APPROVED** [review]

**Comment by basil** [comment]
Moving this out of draft since 2.477 appears to be on track for LTS selection and I just performed a release of this repository. We could merge this now and then just wait about a month to release it until 2.477.1 is released. I have no strong preference, though, so if we want to wait to merge this PR, that is fine with me as well.

**Review by timja — APPROVED** [review]

**Review by jglick — APPROVED** [review]
It seems useful to merge & release now, for the benefit of plugins which were obliged to switch to a 2.477 baseline for whatever reason.

**Comment by scherler** [comment]
@jglick @basil yes please release, I am running into the missing new release for developing things

**Comment by jglick** [comment]
Thanks for release @timja!

File: pom.xml [hunk 4]
```diff
@@ -596,57 +589,6 @@
...
-                        ,
-                        Require upper bound dependencies error for org.ow2.asm:asm-util:9.6 paths to dependency are:
-                        +-org.jenkins-ci.plugins:jacoco:3.3.6
-                          +-io.jenkins.plugins:asm-api:9.7-33.v4d23ef79fcc8
-                            +-org.ow2.asm:asm-util:9.6 (managed) ← org.ow2.asm:asm-util:9.7
-                        ]
-
-                      In the long term, this ought to be mitigated by removing ASM from Jenkins core.
-                    -->
-                    <exclude>org.ow2.asm:asm</exclude>
```
Comment (basil) [inline pom.xml:645]: Causes #1032 for reasons I do not (yet) understand.

**Pull request merged**